lib/popcorn/pop_repos.nit

```diff
@@ -331,6 +331,57 @@ class MongoRepository[E: Serializable]
 	end
 end
 
+# Base serializable entity that can go into a JsonRepository
+#
+# Provide boiler plate implementation of all object serializable to json.
+#
+# `id` is used as a primary key for `find_by_id`.
+#
+# Subclassing RepoObject makes it easy to create a serializable class:
+# ~~~
+# import popcorn::pop_repos
+#
+# class Album
+#	super RepoObject
+#	serialize
+#
+#	var title: String
+#	var price: Float
+# end
+# ~~~
+#
+# Do not forget the `serialize` annotation else the fields will not be serialized.
+#
+# It is also possible to redefine the `id` primary key to use your own:
+# ~~~
+# import popcorn::pop_repos
+#
+# class Order
+#	super RepoObject
+#	serialize
+#
+#	redef var id = "order-{get_time}"
+#
+#	# ...
+#
+# end
+# ~~~
+abstract class RepoObject
+	super Jsonable
+	serialize
+
+	# `self` unique id.
+	#
+	# This attribute is serialized under the key `_id` to be used
+	# as primary key by MongoDb
+	var id: String = (new MongoObjectId).id is serialize_as "_id"
+
+	redef fun to_s do return id
+	redef fun ==(o) do return o isa SELF and id == o.id
```

Comment:
maybe document this since the semantic the the `==` might be confusing when comparing two objects with the same id but different content (because one keep a reference on a old object that was updated in the db)